### PR TITLE
First draft Screenshot testing

### PR DESCRIPTION
**Pull request by sergio-sastre**
Just for you to follow the structure. Need to elaborate more
@v1sar You can take a look to follow suit for ui Testing :)

### CODE REVIEW

File: docs/content/basics/screenshot_testing.md [hunk 1]
```diff
@@ -1,5 +1,49 @@
...
+
+--- more to write here
+
+## Challenges
+--- Draft 
+1. Flakiness (name them)
+2. Emulator configuration in all parts of involved (elaborate)
+
+## Frameworks
+--- Draft
```
Comment (sergio-sastre) [inline docs/content/basics/screenshot_testing.md:35]: I am planning to give a short overview on the special features and drawbacks of the frameworks. I propose we go deep within a repo were we can actually allow people to play around (not as part of the MVP)

File: docs/content/basics/screenshot_testing.md [hunk 2]
```diff
@@ -1,5 +1,86 @@
...
-To be done
+## What it is
+Screenshot testing (also called Snapshot testing) has been in the Android world since 8th October 2015, when Facebook open sourced the first version of their snapshot testing library.
+
+They are a special type of UI tests that inflate a view, take a screenshot of it, and compare it to an already stored image taken as reference. This reference is considered the source of truth: it depicts how the Ui must be displayed, pixel by pixel.
+If the generated snapshot file from the test differs from the reference, the test fails, passes otherwise.
+
+## Motivation
+Screenshot tests are easy to write and maintain and run fast (≈ 1 sec per test), what makes them affordable to execute upon PRs.
+They focus on detecting visual bugs that standard UiAutomator and Espresso cannot, namely
```
Comment (NoNews) [inline docs/content/basics/screenshot_testing.md:11]: Let's highlight `Espresso` and `UiAutomator`

File: docs/content/basics/screenshot_testing.md [hunk 3]
```diff
@@ -1,5 +1,86 @@
...
+3. Execute "record" locally to generate the corresponding snapshot files and open the corresponding PR containing them.
+4. Some teammates review the snapshots to confirm that they reflect the state of the view we are expecting. Once approved and merged, this Snapshot is taken as source of truth for future verifications of the tests.
+5. Verify the snapshots periodically (e.g. on every PR, once a day, etc.). Ideally you have a CI with a job configured to "verify" them. This means, it executes the snapshot tests with the most up-to-date code, and takes a screenshot. After that, it compares this screenshot with the last one uploaded to the CI or the one being pushed in the PR. If they both match, the test passes, fails otherwise.
+
+
+## Challenges
+Unfortunately, Screenshot tests also come with their own problems. 
+
+1. Flakiness: Same as with Ui tests, Snapshot test face flakiness problems. On one hand, we face similar problems to Ui Testing. Most of the issues described in the [flakiness](../practices/flakiness.md) section also apply to Ui tests.
```
Comment (NoNews) [inline docs/content/basics/screenshot_testing.md:42]: `section also apply to Ui tests`

->

`section also apply to Screenshot tests`?

File: docs/content/basics/screenshot_testing.md [hunk 4]
```diff
@@ -1,5 +1,86 @@
...
+We'll describe them in detail in their own section in the future.
+
+2. Emulator configuration in all parts involved (*Instrumented Snapshot tests only*)
+     - Emulators freezing on the CI when idle for a long time. The best practice is the same as with Ui tests: close them right after running the tests
+     - Synchronizing emulators start up before running tests. Instrumented Snapshot tests run on every device adb can detect. For that, we need to start all the emulators and wait till they are ready. We can achieve this by using tools like [swarmer] (https://github.com/gojuno/swarmer) or the new official [Gradle Managed Virtual Devices] (https://developer.android.com/studio/preview/features#gmd)
+     - Avoid Insufficient Storage errors by starting adb emulators with the `--wipe_data` option
+
+## Frameworks
+1. Instrumented
+     - [Facebook Screenshot tests](https://github.com/facebook/screenshot-tests-for-android): First screenshot testing framework on Android.
```
Comment (NoNews) [inline docs/content/basics/screenshot_testing.md:65]: For the future: would be great to have descriptions here and review of such tools from our side.
As an engineer, I want to use the best and I want to see some advise

File: docs/content/basics/screenshot_testing.md [hunk 5]
```diff
@@ -1,5 +1,86 @@
...
+     - [Testify](https://github.com/Shopify/android-testify#readme): Shopify framework. Last one open sourced
+
+2. JVM
+     - [Paparazzi](https://github.com/cashapp/paparazzi): Square framework
+   
+Unlike Ui tests, it is not possible to write shared tests
+
+## Conclusion
+1. Although it has become very popular recently, screenshot testing has been possible on Android since 2015.
+2. Differs from Ui testing on its purpose: Screenshot testing focuses on the visuals, how the views are displayed under different configurations.
```
Comment (NoNews) [inline docs/content/basics/screenshot_testing.md:78]: we can link already written article about it

**Review by NoNews — APPROVED** [review]

File: docs/content/basics/screenshot_testing.md (hunk 4 above)
Comment (sergio-sastre) [reply docs/content/basics/screenshot_testing.md:65]: Definitely. But I would say out of the scope for the MVP. That deserves and article and repo on its own, and I do not know all the frameworks good enough yet

**Pull request merged**